contracts/ROSCA.sol

```diff
@@ -0,0 +1,218 @@
+    */
+  function ROSCA(
+    uint16 roundPeriodInDays_,
+    uint contributionSize_,
+    uint8 minParticipants_,
+    uint startTime_,
+    uint16 serviceFeeInThousandths_)
+  {
+    if (roundPeriodInDays < MIN_ROUND_PERIOD_IN_DAYS || roundPeriodInDays > MAX_ROUND_PERIOD_IN_DAYS) throw;
+    roundPeriodInDays = roundPeriodInDays_;
+    if (contributionSize_ < MIN_CONTRIBUTION_SIZE || contributionSize_ > MAX_CONTRIBUTION_SIZE) throw;
+    contributionSize = contributionSize_;
+
+
+    if (minParticipants_ < MINIMUM_PARTICIPANTS) throw; // there should be at least 2 people to make a group
+    minParticipants = minParticipants_;
+
+    if (startTime_ < (now + MINIMUM_TIME_BEFORE_ROSCA_START)) throw;
+    startTime = startTime_;
+
+    if (serviceFeeInThousandths_ > MAX_FEE_IN_THOUSANDTHS) throw;
+    serviceFeeInThousandths = serviceFeeInThousandths_;
+
+
+    // set foreman
+    foreman = msg.sender;
+
+    // register foreman as a member of ROSCA
+    addMember(msg.sender);
+  }
+
+  function addMember(address newMember)
+    internal
+  {
+    members[newMember] = User({paid: 0 , contributed: 0, alive: true, pendingWithdrawl: 0});
+    membersAddresses.push(newMember);
+  }
+
+  /** startRound() check if the round has ended, if so, get the winner of the currentRound's pot
+    * and add the amount to the winner's pending withdrawal
+    * if there were no bid during the round, address selected at semi-random
+    *
+    * if currentRound = 0(ROSCA hasnt started), no winner is declared
+    */
+
+  function startRound() {
+    uint roundStartTime = startTime + (uint(currentRound)  * (uint(roundPeriodInDays) * 1 days));
+    if (now < roundStartTime || membersAddresses.length < minParticipants)
+      throw;
+
+    if (currentRound != 0) {
+      if (winnerAddress == 0) { // only true when there is no bidder in this round
+        // there is no bid in this round so find an unpaid address for this epoch
+        uint semi_random = now % membersAddresses.length;
+        for (uint i = 0; i < membersAddresses.length; i++) {
+          if(members[membersAddresses[(semi_random + i) % membersAddresses.length]].paid < EPOCH_TO_RUN)
+            winnerAddress = membersAddresses[semi_random + i];
+          break;
+        }
+      }
+      if(members[winnerAddress].paid == EPOCH_TO_RUN) throw;
+      members[winnerAddress].pendingWithdrawl += lowestBid;
+      members[winnerAddress].paid++;
+      LogRoundFundsReleased(winnerAddress, lowestBid);
+    }
+    if (currentRound < membersAddresses.length) {  // reset variables related to bidding
+      lowestBid = contributionSize * membersAddresses.length + 1;
+      winnerAddress = 0;
+
+      currentRound++;
+    }
+  }
+
+  /**
+   * Anyone not already a member of ROSCA can request to join and they'll be put into
+   * pendingJoinRequest until foreman accepts requests or ROSCA has started
+   */
+  function joinRequest()
+    onlyBeforeStart
+  {
```

Comment:
opening brace on same line, same in functions below. Might want to put the entire function header in a single line, if it fits.
Same for functions below.